Utilities/Thread.cpp

```diff
@@ -1823,8 +1824,61 @@ void thread_ctrl::detect_cpu_layout()
 	{
 		g_native_core_layout.store(native_core_arrangement::amd_ccx);
 	}
+	else if (system_id.find("Intel") != std::string::npos)
+	{
+#ifdef _WIN32
+		const LOGICAL_PROCESSOR_RELATIONSHIP relationship = LOGICAL_PROCESSOR_RELATIONSHIP::RelationProcessorCore;
+		DWORD buffer_size = 0;
+
+		// If buffer size is set to 0 bytes, it will be overwritten with the required size
+		if (GetLogicalProcessorInformationEx(relationship, nullptr, &buffer_size))
+		{
+			LOG_ERROR(GENERAL, "GetLogicalProcessorInformationEx returned 0 bytes");
+			return;
+		}
+		DWORD errorcode = GetLastError();
```

Comment:
No space before this block but double spacing after it looks unbalanced. Otherwise good job, thanks for the HT implementation. I had given up on it because it seemed to be platform specific and 'unclean' but looking at it now it doesnt look too bad.  Also, consider a vector or other wrapper for the buffer allocation instead of raw malloc to keep C++ fans happy.

Reply:
I'd prefer having it this way around as the underlying API seems to be designed for malloc or LocalAlloc. Raw malloc shoudn't cause any issues as the code doesn't throw anything and even in the worst case it's only run once.

The double space was there because of a extra line I removed, I'll fix that when I remove the print statements.

Reply:
Its not because of throwing, its just for wrapping cleanup. Like I said, its more for C++ guys really, the API doesn't care how you got the memory. I don't much care either way, but I'm not the only reviewer either.

Reply:
Just do a vector of bytes. `std::vector<u8> buffer(buffer_size);`

Reply:
That felt like something you'd do in Java, but it works and hopefully makes C++ people happy too.. Thanks for the guidance!